internal/tracer.go

```diff
@@ -62,16 +58,23 @@ func (t tracingWriter) Set(key, val string) {
 //		the header and puts it in the Context object. Does not start a new span
 //		as that is started outside when the workflow is actually executed
 type tracingContextPropagator struct {
+	logger *zap.Logger
 	tracer opentracing.Tracer
 }
 
+// NewTracingContextPropagator returns new tracing context propagator object
+func NewTracingContextPropagator(logger *zap.Logger, tracer opentracing.Tracer) ContextPropagator {
+	return &tracingContextPropagator{logger, tracer}
+}
+
 func (t *tracingContextPropagator) Inject(
 	ctx context.Context,
 	hw HeaderWriter,
 ) error {
 	// retrieve span from context object
 	span := opentracing.SpanFromContext(ctx)
 	if span == nil {
+		t.logger.Info("could not retrieve span from context")
```

Comment:
Potential noisy logs?

Reply:
Maybe, I am not sure. i am going to leave it around for now and remove if I think its too noisy. But if someone has created a tracer and passed it into the client these should not fire and indicate a bad configuration somewhere else on users part.